app/models/merchant.rb

```diff
@@ -36,4 +36,8 @@ def current_status
       "Enabled"
     end
   end
+
+  def self.top_five_merchants
+    joins(invoices: :transactions).where(transactions: { result: "success"}).select("merchants.id, merchants.name, SUM(invoice_items.unit_price * invoice_items.quantity) as total").group(:id).order("total desc").limit(5)
```

Comment:
Huh. Researched this a bit. Interesting that sum calculates products in postgresql